synapse/rest/client/room.py

```diff
@@ -61,6 +64,69 @@
 logger = logging.getLogger(__name__)
 
 
+T_RoomSize = TypeVar("T_RoomSize", bound="_RoomSize")
+
+
+class _RoomSize(Enum):
+    """
+    Enum to differentiate sizes of rooms. This is a pretty good aproximation
+    about how hard it will be to get events in the room. We could also look at
+    room "complexity".
+    """
+
+    # This doesn't necessarily mean the room is a DM, just that there is a DM
+    # amount of people there.
+    DM_SIZE = "direct_message_size"
+    SMALL = "small"
+    SUBSTANTIAL = "substantial"
+    LARGE = "large"
+
+    @classmethod
+    def from_member_count(cls: Type[T_RoomSize], member_count: int) -> "_RoomSize":
+        if member_count <= 2:
+            return _RoomSize.DM_SIZE
+        elif member_count < 100:
+            return _RoomSize.SMALL
+        elif member_count < 1000:
+            return _RoomSize.SUBSTANTIAL
+        else:
+            return _RoomSize.LARGE
```

Comment:
Why is `T_RoomSize` here / what is it doing for us?

Should this be a static method instead and remove the `cls` reference, since unless I'm just bad at reading, I can't see you actually using it?

```suggestion
    @staticmethod
    def from_member_count(member_count: int) -> "_RoomSize":
        if member_count <= 2:
            return _RoomSize.DM_SIZE
        elif member_count < 100:
            return _RoomSize.SMALL
        elif member_count < 1000:
            return _RoomSize.SUBSTANTIAL
        else:
            return _RoomSize.LARGE
```
(& remove defn of `T_RoomSize = ..` above)

Reply:
I just didn't know exactly how to accomplish this in Python. This looks much cleaner 👍 Thanks for the example!